server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -814,6 +840,198 @@ private Scheduler.Cancellable fetchClusterFormationInfo(
+                    TransportRequestOptions.timeout(transportTimeout),
+                    new ActionListenerResponseHandler<>(
+                        ActionListener.runBefore(fetchCoordinationDiagnosticsListener, () -> Releasables.close(releasable)),
+                        CoordinationDiagnosticsAction.Response::new
+                    )
+                );
+            }
+        }, e -> {
+            logger.warn("Exception connecting to master masterEligibleNode", e);
+            responseConsumer.accept(new RemoteMasterHealthResult(masterEligibleNode, null, e));
+        });
+
+        fetchCoordinationDiagnosticsListener.whenComplete(response -> {
+            long endTime = System.nanoTime();
+            logger.trace(
+                "Received master stability result from {} in {}",
+                masterEligibleNode,
+                TimeValue.timeValueNanos(endTime - startTime)
+            );
+            /*
+             * It is possible that the task has been cancelled at this point, but it does not really matter. In that case this result
+             * will be ignored and soon garbage collected.
+             */
+            responseConsumer.accept(new RemoteMasterHealthResult(masterEligibleNode, response.getCoordinationDiagnosticsResult(), null));
+        }, e -> {
+            logger.warn("Exception in master stability request to master masterEligibleNode", e);
+            responseConsumer.accept(new RemoteMasterHealthResult(masterEligibleNode, null, e));
+        });
+
+        return transportService.getThreadPool().schedule(() -> {
+            if (masterEligibleNode == null) {
+                /*
+                 * This node's PeerFinder hasn't yet discovered the master-eligible nodes. By notifying the responseConsumer with a null
+                 * value we effectively do nothing, and allow this request to be recheduled by rescheduleDiagnosticsFetchConsumer().
+                 */
+                responseConsumer.accept(null);
+            } else {
+                Version minSupportedVersion = Version.V_8_4_0;
+                if (masterEligibleNode.getVersion().onOrAfter(minSupportedVersion) == false) { // This was introduced in 8.4.0
+                    logger.trace(
+                        "Cannot get cluster coordination info for {} because it is at version {} and {} is required",
+                        masterEligibleNode,
+                        masterEligibleNode.getVersion(),
+                        minSupportedVersion
+                    );
+                } else {
+                    /*
+                     * Since this block is not synchronized it is possible that this task is cancelled between the check above and when the
+                     * code below is run, but this is harmless and not worth the additional synchronization in the normal case. In that case
+                     * the connection will just be closed and the transport request will not be made.
+                     */
```

Comment:
This is a bit misleading as the check above is either 
``` 
if (masterEligibleNode == null) {
```
or
```
if (masterEligibleNode.getVersion().onOrAfter(minSupportedVersion) == false) { 
```
I think we should rephrase this comment

Reply:
What I was trying to say is better said in rescheduleDiagnosticsFetchConsumer, so removing this comment.